File: Loenn/entities/tumbleweedCluster.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local tumbleweedCluster = {}

tumbleweedCluster.name = "MaggyHelper/TumbleweedCluster"
tumbleweedCluster.depth = -50
tumbleweedCluster.fieldInformation = {
    rollSpeed = {
        minimumValue = 0
    },
    pushForce = {
        minimumValue = 0
    },
    tumbleweedCount = {
        minimumValue = 1
    },
    bounceChance = {
        minimumValue = 0,
        maximumValue = 1
    }
}
tumbleweedCluster.placements = {
    {
        name = "normal",
        data = {
            rollSpeed = 180,
            pushForce = 100,
            tumbleweedCount = 3,
            bounceChance = 0.3
        }
    }
}

function tumbleweedCluster.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/tumbleweed/idle", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function tumbleweedCluster.selection(room, entity)
    return utils.rectangle(entity.x - 30, entity.y - 20, 60, 40)
end

return tumbleweedCluster

File: Ahorn/entities/TumbleweedCluster.jl
module TumbleweedCluster

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/TumbleweedCluster" TumbleweedCluster(x::Integer, y::Integer, bounceChance::Number=0.3, pushForce::Integer=100, rollSpeed::Integer=180, tumbleweedCount::Integer=3)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(TumbleweedCluster)
)

function Ahorn.selection(entity::TumbleweedCluster)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::TumbleweedCluster, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
